lib/application_insights/channel/contracts/data.rb

```diff
@@ -1,48 +1,9 @@
 require_relative 'json_serializable'
 
-module ApplicationInsights
-  module Channel
-    module Contracts
-      # Data contract class for type Data.
-      class Data < JsonSerializable
-        # Initializes a new instance of the Data class.
-        def initialize(options={})
-          defaults = {
-            'baseType' => nil,
-            'baseData' => nil
-          }
-          values = {
-            'baseData' => nil
-          }
-          super defaults, values, options
-        end
-        
-        # Gets the base_type property.
-        def base_type
-          @values.fetch('baseType') { 
-            @values['baseType'] = nil
-          }
-        end
-        
-        # Sets the base_type property.
-        def base_type=(value)
-          if value == @defaults['baseType']
-            @values.delete 'baseType' if @values.key? 'baseType'
-          else
-            @values['baseType'] = value
-          end
-        end
-        
-        # Gets the base_data property.
-        def base_data
-          @values['baseData']
-        end
-        
-        # Sets the base_data property.
-        def base_data=(value)
-          @values['baseData'] = value
-        end
-      end
-    end
+module ApplicationInsights::Channel::Contracts
+  class Data
+    include JsonSerializable
+
+    attributes :baseType, :baseData
```

Comment:
I understand your concern, but I believe that every developer reads the docs or tests to see how a gem works. Actually you must read it to understand how to send your metrics, because it is straightforward.

If you don't like this implementation we can pair program to find a better solution!